src/AutoMapper/Mappers/TypeHelper.cs

```diff
@@ -88,6 +94,20 @@ private static Type GetIEnumerableType(Type enumerableType)
                 return null;
             }
         }
+        private static Type GetIDictionaryType(Type enumerableType)
+        {
+            try
+            {
+                return enumerableType.GetTypeInfo().ImplementedInterfaces.FirstOrDefault(t => t.Name == "IDictionary`2");
+            }
+            catch (AmbiguousMatchException)

```

Comment:
That doesn't look good. There is an extension method somewhere ImplementsGenericInterface or smth.